pkg/security/probe/probe_linux.go

```diff
@@ -484,6 +484,38 @@ func (p *Probe) onEventLost(perfMapName string, perEvent map[string]uint64) {
 	}
 }
 
+// setProcessContext set the process context, should return false if the event shouldn't be dispatched
+func (p *Probe) setProcessContext(eventType model.EventType, event *model.Event) bool {
+	entry, isResolved := p.fieldHandlers.ResolveProcessCacheEntry(event)
+	if !eventWithNoProcessContext(eventType) {
+		if !isResolved {
+			event.Error = &ErrNoProcessContext{Err: errors.New("process context not resolved")}
+		} else if !entry.HasValidLineage() {
+			event.Error = &ErrProcessBrokenLineage{PIDContext: entry.PIDContext}
+			p.resolvers.ProcessResolver.CountBrokenLineage()
+		}
```

Comment:
Should we return false here to avoid dispatching a broken event?

Reply:
We reject the broken event at the CWS consumer side as we don't want to reject the event for the other consumers that may not be impacted by broken lineage. https://github.com/DataDog/datadog-agent/blob/main/pkg/security/rules/engine.go#L425

Reply:
I see, thanks ! :)